Backport of:

From 2257d1eadd02d89d225fce21013a1219d221dc7d Mon Sep 17 00:00:00 2001
From: dirk <dirk@aa41f4f7-0bf4-0310-aa73-e5a19afd5a74>
Date: Tue, 16 Dec 2014 12:17:20 +0000
Subject: [PATCH] Thread limit should be at least 1.

Limit thread when thread limit is 0. It is a logic error that could lead to ressource exhaustion.

git-svn-id: https://subversion.imagemagick.org/subversion/ImageMagick/branches/ImageMagick-6@17282 aa41f4f7-0bf4-0310-aa73-e5a19afd5a74
origin:  http://trac.imagemagick.org/changeset/17282
---
 magick/resource.c | 2 ++
 1 file changed, 2 insertions(+)

Index: imagemagick-6.6.9.7/magick/resource.c
===================================================================
--- imagemagick-6.6.9.7.orig/magick/resource.c	2016-11-15 13:08:18.483636679 -0500
+++ imagemagick-6.6.9.7/magick/resource.c	2016-11-15 13:11:27.841948802 -0500
@@ -1140,6 +1140,8 @@
       if (value != (char *) NULL)
         resource_info.thread_limit=MagickMin(limit,StringToSizeType(value,
           100.0));
+      if (resource_info.thread_limit == 0)
+        resource_info.thread_limit=1;
       SetOpenMPMaximumThreads((int) resource_info.thread_limit);
       break;
     }
